src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs

```diff
@@ -116,16 +124,44 @@ internal async Task<IEnumerable<IProjectVersionedValue<RestoreData>>> RestoreAsy
             };
         }
 
+        // means either it has never receive computed value from configured project yet, or the computed data is out of dated.
+        private void CheckPendingNomination(IProjectVersionedValue<PackageRestoreUnconfiguredInput> e)
+        {
+            Assumes.Present(_project.Services.ActiveConfiguredProjectProvider);
+            Assumes.Present(_project.Services.ActiveConfiguredProjectProvider.ActiveConfiguredProject);
+
+            // HasPendingNomination means either it has never receive computed value from configured project yet.
+            bool neverReceivedComputedValue = e.Value.ConfiguredInputs.Count == 0 ? true : false;
+
+            // PackageRestoreDataSource can compare the version number it gets and the ConfiguredProject.Version.
+            // If the version it gets is older than the current project state, it means it is getting out of dated data.
+            // HasPendingNomination will be true.
+            ConfiguredProject? activeConfiguredProject = _project.Services.ActiveConfiguredProjectProvider.ActiveConfiguredProject;
+
+            bool computedDataItOutOfDate = false;
+            foreach (var input in e.Value.ConfiguredInputs)
+            {
+                computedDataItOutOfDate = string.CompareOrdinal(input.ProjectConfiguration.Name, activeConfiguredProject.ProjectConfiguration.Name)==0 &&
+                    input.ConfiguredProjectVersion.IsLaterThan(activeConfiguredProject.ProjectVersion) ? true : computedDataItOutOfDate;
+            }
+
+            _pendingNomination = neverReceivedComputedValue || computedDataItOutOfDate;
+        }
+
         private async Task<bool> RestoreCoreAsync(ProjectRestoreInfo restoreInfo)
         {
             // Restore service always does work regardless of whether the value we pass 
             // them to actually contains changes, only nominate if there are any.
             byte[] hash = RestoreHasher.CalculateHash(restoreInfo);
 
             if (_latestHash != null && Enumerable.SequenceEqual(hash, _latestHash))
+            {
+                _norminateNextRestore = false;
                 return true;
+            }
 
             _latestHash = hash;
+            _norminateNextRestore = true;
```

Comment:
A comment here would be useful, explaining why this is set. Or alternatively, a comment on the field itself explaining how it is used.

Reply:
not needed anymore